manifests/cli.pp

```diff
@@ -0,0 +1,92 @@
+# = Pulpcore command line interface
+#
+# This class installs the Pulpcore command line interface (CLI).
+#
+# === Parameters:
+#
+# $pulpcore_url::       URL on which Pulpcore runs
+#
+# $username::           Username for authentication
+#
+# $password::           Password for authentication
+#
+# $cert::               Client certificate for authentication
+#
+# $key::                Client key for authentication
+#
+# === Advanced parameters:
+#
+# $manage_root_config:: Whether to manage /root/.config/pulp configuration.
+#
+# $api_root::           Absolute API base path on server (not including 'api/v3/')
+#
+# $verify_ssl::         Verify SSL connection
+#
+# $dry_run::            Trace commands without performing any unsafe HTTP calls
+#
+# $version::            pulp-cli package version, it's passed to ensure parameter of package resource
+#                       can be set to specific version number, 'latest', 'present' etc.
+#
+class pulpcore::cli (
+  Optional[Stdlib::HTTPUrl] $pulpcore_url = undef,
+  String $version = 'installed',
+  Optional[String] $username = undef,
+  Optional[String] $password = undef,
+  Optional[String] $cert = undef,
+  Optional[String] $key = undef,
+  Optional[String] $api_root = undef,
+  Boolean $verify_ssl = true,
+  Boolean $dry_run = true,
+  Boolean $manage_root_config = true,
+) {
+  package { 'pulp-cli':
+    ensure => $version,
+  }
+  -> file { '/etc/pulp/cli.toml':
+    ensure  => file,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+    content => epp(
+      'pulpcore/cli-config.toml.epp',
+      {
+        base_url   => $pulpcore_url,
+        api_root   => $api_root,
+        verify_ssl => $verify_ssl,
+        dry_run    => $dry_run,
+      }
+    ),
+  }
+
+  if $manage_root_config and (($username and $password) or ($cert and $key)) {
```

Comment:
So in `pulp-cli` the validation is a bit more involved: https://github.com/pulp/pulp-cli/blob/aa50bc63e949e7f413bce6bc10d323d119e61b09/pulpcore/cli/common/openapi.py#L49-L62

I think for this module, it would be sufficient to check:
* (username OR password) AND (cert OR key) → error
* password AND !username → error
* key AND !cert → error

Everything else is imho valid (like setting username via config, but using CLI `--password` for the password, or providing a cert/key combo in the cert parameter).